[BUG]

### Information
1. **Apktool Version (`apktool -version`)** - 3.0.1 and 3.0.2 (both fail). 2.12.0 works.
2. **Operating System (Mac, Linux, Windows)** - Mac
3. **APK From? (Playstore, ROM, Other)** - Other (built locally with AGP 8.11.2 from a stock empty Android Studio project; also reproduced on a Play Store APK).
4. **Java Version (`java --version`)** - openjdk 17.0.2

### Stacktrace/Logcat

```
I: Using Apktool 3.0.2 on demo-app.apk with 8 threads
 I: Smaling smali folder into classes.dex...
 I: Smaling smali_classes2 folder into classes2.dex...
 I: Smaling smali_classes3 folder into classes3.dex...
 I: Building resources with aapt2...
 W: .../res/layout/repro_gravity.xml:2: error: '0x0' is incompatible with attribute gravity (attr) flags [bottom=80, center=17, center_horizontal=1, center_vertical=16, clip_horizontal=8, clip_vertical=128, end=8388613, fill=119, fill_horizontal=7, fill_vertical=112, left=3, right=5, start=8388611, top=48].
 W: error: failed linking file resources.
 Exception in thread "main" brut.androlib.exceptions.AndrolibException: brut.common.BrutException: Execution failed (exit code = 1): [/var/folders/g_/nzcfqxhx2yvg4qx118_k_myw0000gn/T/aapt2_35783033096881524703791754876596627910.tmp, link, -o, /var/folders/g_/nzcfqxhx2yvg4qx118_k_myw0000gn/T/APKTOOL8066699186008165733.tmp, --manifest, .../AndroidManifest.xml, --min-sdk-version, 32, --target-sdk-version, 36, --version-code, 1, --version-name, 1.0, --package-id, 127, --no-auto-version, --no-version-vectors, --no-version-transitions, --no-resource-deduping, --no-compile-sdk-metadata, --warn-manifest-validation, -I, .../Library/apktool/framework/1.apk, .../build/resources.zip]
   at brut.androlib.res.AaptInvoker.invoke(SourceFile:186)
   at brut.androlib.ApkBuilder.buildResources(SourceFile:357)
   at brut.apktool.Main.main(SourceFile:92)
 Caused by: brut.common.BrutException: Execution failed (exit code = 1): [/var/folders/g_/nzcfqxhx2yvg4qx118_k_myw0000gn/T/aapt2_35783033096881524703791754876596627910.tmp, link, -o, /var/folders/g_/nzcfqxhx2yvg4qx118_k_myw0000gn/T/APKTOOL8066699186008165733.tmp, --manifest, .../AndroidManifest.xml, --min-sdk-version, 32, --target-sdk-version, 36, --version-code, 1, --version-name, 1.0, --package-id, 127, --no-auto-version, --no-version-vectors, --no-version-transitions, --no-resource-deduping, --no-compile-sdk-metadata, --warn-manifest-validation, -I, .../Library/apktool/framework/1.apk, .../build/resources.zip]
   at brut.util.OS.exec(SourceFile:147)
   at brut.androlib.res.AaptInvoker.invoke(SourceFile:183)
   ... 2 more
```

### Steps to Reproduce
Minimal layout that triggers it (only addition to a stock empty Android Studio project):
```
<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
 android:layout_width="match_parent"
 android:layout_height="wrap_content"
 android:gravity=""
 android:orientation="vertical" />
```

* AGP/AAPT2 compiles this fine and stores gravity as flags = 0x0 (no bits set). The APK installs and runs on Android.

This works:
1. ```java -jar apktool_2.12.0.jar d demo.apk -o decoded -f```
2. ```java -jar apktool_2.12.0.jar b decoded -o rebuilt.apk``` # OK

This fails:
1. ```java -jar apktool_3.0.2.jar d demo.apk -o decoded -f```
2. ```java -jar apktool_3.0.2.jar b decoded -o rebuilt.apk``` # FAIL (stacktrace above)

Decoded value also differs between versions for the same input APK:
• 2.12.0 → android:gravity=""
• 3.0.2 → android:gravity="0x0"

### Frameworks
N/A (not an OEM ROM APK).

### APK
If this APK can be freely shared, please upload/attach a link to it.

### Questions to ask before submission
1. Have you tried `apktool d`, `apktool b` without changing anything?
    Yes — apktool d then apktool b with no changes; fails on 3.0.x, succeeds on 2.12.0.

2. If you are trying to install a modified apk, did you resign it?
    N/A — failing at the rebuild step, before signing.

3. Are you using the latest apktool version?
reproduced on the latest 3.0.2.

### Question
Source APK builds and runs fine (Google's AAPT2 accepts gravity="" and emits 0x0), so the input doesn't seem invalid.
Is this an intended stricter validation in 3.0.x, or a regression vs 2.12.0?

--- a/brut.apktool/apktool-lib/src/main/java/brut/androlib/res/data/StyledString.java
+++ b/brut.apktool/apktool-lib/src/main/java/brut/androlib/res/data/StyledString.java
@@ -98,5 +98,10 @@ public class StyledString implements CharSequence {
         public int getLastChar() {
             return mLastChar;
         }
+
+        @Override
+        public String toString() {
+            return String.format("Span{tag=%s, firstChar=%s, lastChar=%s}", mTag, mFirstChar, mLastChar);
+        }
     }
 }

--- a/brut.apktool/apktool-lib/src/main/java/brut/androlib/res/table/value/ResAttribute.java
+++ b/brut.apktool/apktool-lib/src/main/java/brut/androlib/res/table/value/ResAttribute.java
@@ -146,6 +146,11 @@ public class ResAttribute extends ResBag {
         public ResPrimitive getValue() {
             return mValue;
         }
+
+        @Override
+        public String toString() {
+            return String.format("Symbol{key=%s, value=%s}", mKey, mValue);
+        }
     }
 
     public void addValueType(int valueType) {

--- a/brut.apktool/apktool-lib/src/main/java/brut/androlib/res/table/value/ResBag.java
+++ b/brut.apktool/apktool-lib/src/main/java/brut/androlib/res/table/value/ResBag.java
@@ -63,6 +63,11 @@ public abstract class ResBag extends ResValue implements ValuesXmlSerializable {
         public ResItem getValue() {
             return mValue;
         }
+
+        @Override
+        public String toString() {
+            return String.format("RawItem{key=0x%08x, value=%s}", mKey, mValue);
+        }
     }
 
     public void resolveKeys() throws AndrolibException {

--- a/brut.apktool/apktool-lib/src/main/java/brut/androlib/res/table/value/ResEnum.java
+++ b/brut.apktool/apktool-lib/src/main/java/brut/androlib/res/table/value/ResEnum.java
@@ -109,9 +109,13 @@ public class ResEnum extends ResAttribute {
             }
         }
 
+        // Stop early if the value is missing a symbol.
         if (symbolsCount == 0) {
-            symbols = null;
-        } else if (symbolsCount < symbols.length) {
+            mSymbolsCache.put(data, null);
+            return null;
+        }
+
+        if (symbolsCount < symbols.length) {
             symbols = Arrays.copyOf(symbols, symbolsCount);
         }
 

--- a/brut.apktool/apktool-lib/src/main/java/brut/androlib/res/table/value/ResFlags.java
+++ b/brut.apktool/apktool-lib/src/main/java/brut/androlib/res/table/value/ResFlags.java
@@ -139,6 +139,12 @@ public class ResFlags extends ResAttribute {
                 }
             }
 
+            // Stop early if any of the flags are missing a symbol.
+            if (mask != data) {
+                mSymbolsCache.put(data, null);
+                return null;
+            }
+
             // Filter out redundant flags.
             if (symbolsCount > 2) {
                 Symbol[] filtered = new Symbol[symbolsCount];
@@ -170,9 +176,7 @@ public class ResFlags extends ResAttribute {
             }
         }
 
-        if (symbolsCount == 0) {
-            symbols = null;
-        } else if (symbolsCount < symbols.length) {
+        if (symbolsCount < symbols.length) {
             symbols = Arrays.copyOf(symbols, symbolsCount);
         }
 

--- a/brut.apktool/apktool-lib/src/main/java/brut/androlib/res/table/value/ResStyle.java
+++ b/brut.apktool/apktool-lib/src/main/java/brut/androlib/res/table/value/ResStyle.java
@@ -76,6 +76,11 @@ public class ResStyle extends ResBag {
         public ResItem getValue() {
             return mValue;
         }
+
+        @Override
+        public String toString() {
+            return String.format("Item{key=%s, value=%s}", mKey, mValue);
+        }
     }
 
     @Override

